src/k8s/pkg/k8sd/types/node_config.go

```diff
@@ -0,0 +1,7 @@
+package types
+
+type NodeConfig struct {
+	CloudProvider string `mapstructure:"cloud-provider,omitempty"`
```

Comment:
no need to pull in mapstructure - just use `json`

Reply:
We can technically use `json` but wouldn't that mean doing a `Marshal` and `Unmarshal` just to convert between `map[string]string` and `NodeConfig`. Seems a bit inefficient/hacky imo.